elastic/kibana
# [Fleet] allow agent upgrades if patch version is higher than kibana

## Summary

Closes https://github.com/elastic/kibana/issues/168502

Changed the version check to allow agent upgrade if the agent version has a newer patch than kibana.

To verify:
- change kibana locally to return a mock version `8.11.0` [here](https://github.com/elastic/kibana/blob/05bfe53cb3a2fe33ecb9eec4a6fcb19a492aaadf/x-pack/plugins/fleet/public/hooks/use_kibana_version.ts#L17)
- enroll an agent version 8.11.0
- verify that the upgrade is allowed to 8.11.1 and 8.11.2
- verify that the upgrade works

<img width="1282" alt="image" src="https://github.com/elastic/kibana/assets/90178898/0bb0c8ca-ac0f-49c1-b67c-b02d085e7045">
<img width="799" alt="image" src="https://github.com/elastic/kibana/assets/90178898/7be3b64b-43e5-451a-a630-65cfc2607dab">
<img width="1256" alt="image" src="https://github.com/elastic/kibana/assets/90178898/4841bc2a-5238-4854-a63e-27761a02f1e3">

Tested the new agent build version by adding a dummy version to the available_versions API response.
It is showing up for an agent 8.11.1 (same as the mock kibana version):
<img width="775" alt="image" src="https://github.com/elastic/kibana/assets/90178898/fa02c62d-9399-4c2c-8311-412e5fe03a96">



### Checklist
